src/customer/strategies/masterpass-customer-strategy.ts

```diff
@@ -0,0 +1,115 @@
+import { PaymentMethod, PaymentMethodActionCreator } from '../../payment';
+import { MasterpassScriptLoader } from '../../payment/strategies/masterpass';
+import { RemoteCheckoutActionCreator } from '../../remote-checkout';
+import CustomerCredentials from '../customer-credentials';
+import {CustomerInitializeOptions, CustomerRequestOptions} from '../customer-request-options';
+
+import CustomerStrategy from './customer-strategy';
+
+export default class MasterpassCustomerStrategy extends CustomerStrategy {
+    private _paymentMethod?: PaymentMethod;
+
+    constructor(
+        store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _remoteCheckoutActionCreator: RemoteCheckoutActionCreator,
+        private _masterpassScriptLoader: MasterpassScriptLoader
+    ) {
+        super(store);
+    }
+
+    initialize(options: CustomerInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { masterpass: masterpassOptions, methodId } = options;
+
+        if (!masterpassOptions || !methodId) {
+            throw new InvalidArgumentError('Unable to proceed because "options.masterpass" argument is not provided.');
+        }
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+                const cart = state.cart.getCart();
+                const storeConfig = state.config.getStoreConfig();
+
+                if (!cart) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+                if (!storeConfig) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+                }
+                if (!this._paymentMethod || !this._paymentMethod.initializationData.checkoutId) {
+                    throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+                }
+                const container = masterpassOptions.container;
+
+                return this._masterpassScriptLoader.load(this._paymentMethod.config.testMode)
+                    .then(Masterpass => {
+                        const signInButton = this._createSignInButton(container);
+                        if (!this._paymentMethod) {
+                            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+                        }
+                        const payload = {
+                            checkoutId: this._paymentMethod.initializationData.checkoutId,
+                            allowedCardTypes: this._paymentMethod.initializationData.allowedCardTypes,
+                            amount: String(cart.cartAmount.toFixed(2)),
```

Comment:
please directly use toString() as the String constructor called as a function (your first example), will at the end, call the toString()